ddt.py

```diff
@@ -152,25 +159,41 @@ def process_file_data(cls, name, func, file_attr):
     cls_path = os.path.abspath(inspect.getsourcefile(cls))
     data_file_path = os.path.join(os.path.dirname(cls_path), file_attr)
 
-    def _raise_ve(*args):  # pylint: disable-msg=W0613
+    def _raise_not_exists_error(*args):  # pylint: disable-msg=W0613
         raise ValueError("%s does not exist" % file_attr)
 
+    def _raise_need_yaml_error(*args):  # pylint: disable-msg=W0613
+        raise ValueError("%s is a YAML file, please install PyYAML" % file_attr)
+
+    # If file does not exist, provide an error function instead
     if os.path.exists(data_file_path) is False:
         test_name = mk_test_name(name, "error")
-        add_test(cls, test_name, _raise_ve, None)
+        add_test(cls, test_name, _raise_not_exists_error, None)
+        return
+
+    # Load the data as YAML or JSON
+    if data_file_path.endswith((".yml", ".yaml")):
+        # Don't have YAML but want to use YAML file.
+        if not _have_yaml:
+            test_name = mk_test_name(name, "error")
+            add_test(cls, test_name, _raise_need_yaml_error, None)
+            return
+        data = yaml.load(open(data_file_path).read())
     else:
         data = json.loads(open(data_file_path).read())
```

Comment:
It was tempting to fix this to properly close the file. I think I'll let it stay as is in this PR.

The file should be closed. Do you wish to support Python <2.6?